dev-docs/source/GitConfig.rst

```diff
@@ -0,0 +1,164 @@
+           conflictstyle = diff3
+   [rebase]
+           # Automatically reposition !fixup/!squash commits in interactive rebase (useful with git commit --fixup/squash)
+           autosquash = true
+
+Diff/Merge Tools
+----------------
+
+You may find the following diff/merge tools helpful:
+
+.. code-block:: sh
+
+   # Uncomment lines where appropriate for specific OS
+   [diff]
+           # Set up a graphical tool for viewing diffs. I like meld because it looks good and does a good job with directory diffs.
+           tool = meld
+   [difftool]
+           # Disable the prompt when opening the difftool
+           prompt = false
+   [merge]
+           # Specify a graphical tool for resolving merge conflicts. I like diffmerge because you can toggle between the merged file and common ancestor
+           tool = diffmerge
+   [difftool "meld"]
+           external = meld
+           # Windows
+           #cmd = \"C:\\Program Files (x86)\\Meld\\Meld.exe\" "$LOCAL" "$REMOTE"
+           #path = C:/Program Files (x86)/Meld/Meld.exe
+   [mergetool "meld"]
+           trustExitCode = true
+           keepBackup = false
+           # Windows
+           #cmd = \"C:\\Program Files (x86)\\Meld\\Meld.exe\" "$LOCAL" "$BASE" "$REMOTE" "--output=$MERGED"  
+           #path = C:/Program Files (x86)/Meld/Meld.exe
+   [difftool "diffmerge"]
+           # Linux
+           #cmd = diffmerge $LOCAL $REMOTE
+           # Windows
+           #cmd = C:/Program\\ Files/SourceGear/common/DiffMerge/sgdm.exe \"$LOCAL\" \"$REMOTE\"
+   [mergetool "diffmerge"]
+           trustExitCode = true
+           keepBackup = false
+           # Linux
+           #cmd = diffmerge -merge -result=\"$MERGED\" \"$LOCAL\" \"$BASE\" \"$REMOTE\"
+           # Windows
+           #cmd = C:/Program\\ Files/SourceGear/common/DiffMerge/sgdm.exe -merge -result=\"$MERGED\" \"$LOCAL\" \"$BASE\" \"$REMOTE\"
+
+Useful Aliases
+--------------
+
+You may find the following aliases helpful. You can also add your own.
+
+.. code-block:: sh
+
+   [pretty]
+           concise = "%C(yellow)%h%Creset %C(green)[%ar]%Creset %C(auto)%d%Creset %s %C(cyan)<%an>"
+           detail = "%C(yellow)commit %h%Creset%C(auto)%d%Creset%n%C(yellow)Parents: %p%Creset%n%C(cyan)Author: %an <%ae>%Creset%n%C(cyan)        %ai (%ar)%Creset%n%C(green)Commit: %cn <%ce>%Creset%n%C(green)        %ci (%cr)%Creset%n%n%w(79)%s%n%n%b"
+
+   [alias]
+           # Check out a pull request for testing - uncomment and replace <main-remote-name>
+           test-pr = "!f() { git fetch $1 pull/$2/merge:pr/$2-merged && git checkout pr/$2-merged; }; f"
+           # Remove all branches starting pr/
+           test-pr-remove-all = "!f() { git branch | grep pr/ | xargs git branch -D; }; f"
+
+           # Get into the habit of using add --patch to create focused commits
+           ap = add --patch
+           # Amend the latest commit
+           ca = commit --amend
```

Comment:
I use
```
# Amend the latest commit reuse commit message from latest commit 
cah = commit --amend -C HEAD
```
quite a lot for fixing up the latest commit when I realise there's something I missed or I've created work-in-progress commit that I continue with later.